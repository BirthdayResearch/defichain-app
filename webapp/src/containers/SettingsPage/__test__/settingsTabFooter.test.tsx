import React from 'react';
import SettingsTabFooterComponent from '../components/SettingsTabFooter';
import { shallow } from 'enzyme';

describe('SettingsTabFooterComponent', () => {
  it('should check for snapshot if isUnsaved is true', () => {
    const props = {
      saveChanges: () => {},
      isUnsavedChanges: true,
    };
    const wrapper = shallow(<SettingsTabFooterComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should check for snapshot if isUnsaved is true', () => {
    const props = {
      saveChanges: () => {},
      isUnsavedChanges: false,
    };
    const wrapper = shallow(<SettingsTabFooterComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
