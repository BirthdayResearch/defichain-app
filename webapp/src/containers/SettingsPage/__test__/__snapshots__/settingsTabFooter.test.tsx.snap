// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsTabFooterComponent should check for snapshot if isUnsaved is true 1`] = `
<footer
  className="footer-bar"
>
  <Row
    className="justify-content-between align-items-center"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      className="col-auto"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      You have unsaved changes
    </Col>
    <Col
      className="d-flex justify-content-end"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        color="primary"
        disabled={false}
        onClick={[Function]}
        tag="button"
      >
        Save settings
      </Button>
    </Col>
  </Row>
</footer>
`;

exports[`SettingsTabFooterComponent should check for snapshot if isUnsaved is true 2`] = `
<footer
  className="footer-bar"
>
  <Row
    className="justify-content-between align-items-center"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <Col
      className="col-auto"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    />
    <Col
      className="d-flex justify-content-end"
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Button
        color="primary"
        disabled={true}
        onClick={[Function]}
        tag="button"
      >
        Save settings
      </Button>
    </Col>
  </Row>
</footer>
`;
